web/src/main/java/org/cbioportal/web/ClinicalDataController.java

```diff
@@ -38,7 +38,7 @@
     @RequestMapping(value = "/studies/{studyId}/samples/{sampleId}/clinical-data", method = RequestMethod.GET,
             produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Get all clinical data of a sample in a study")
-    public ResponseEntity<List<? extends ClinicalDataSummary>> getAllClinicalDataOfSampleInStudy(
+    public ResponseEntity<List<ClinicalData>> getAllClinicalDataOfSampleInStudy(
             @ApiParam(required = true, value = "Study ID e.g. acc_tcga")
```

Comment:
I guess we are relying on this required specification to prevent null values in the studyId and sampleId. But maybe tests for null at the service layer .. and more importantly the persistence layer .. could make sense, because these layers may be called by other code (validator? performance testing code?) where the programmer passes in a null value (maybe thinking that null means "return all" or something). Thoughts?

For now, I'll approve without null argument checks since these are required arguments at the controller level.